### PR TITLE
Fetch required secrets 751

### PR DESCRIPTION
**Pull request by jonahx**
#### What does this PR do?

- Adds a reusable command class for validating and fetching required secrets (for use in OIDC and elsewhere)
- Adds tests for that class
- Adds a utility class `DeepDouble` (soon to be a gem) that removes a good deal of rspec boilerplate for tests that use doubles, especially nested ones.

#### Any background context you want to provide?

Inspired as part of the OIDC project

#### What ticket does this PR close?

https://github.com/cyberark/conjur/issues/751

#### Link to build in Jenkins (if appropriate)

https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--conjur/detail/fetch-required-secrets-751/5/pipeline/

#### Has the Version and Changelog been updated?

No.

It has unit tests.  Integration tests are N/A.

### CODE REVIEW

File: app/domain/conjur/fetch_required_secrets.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+require 'command_class'
+require 'util/error_class'
+
+module Conjur
+
+  # TODO: really, these errors should be inside FetchRequiredSecrets but that
```
Comment (micahlee) [inline app/domain/conjur/fetch_required_secrets.rb]: If this is technical debt that should be fixed, would you mind adding an issue for it?

File: app/domain/util/stubs/deep_double.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+module Util
+  module Stubs
+    class DeepDouble
+
+      class LookupReturnValue
```
Comment (micahlee) [inline app/domain/util/stubs/deep_double.rb]: How do you feel about splitting `LookupReturnValue` into its own file? It took me a minute to tease apart `DeepDouble` from `LookupReturnValue`, and I imagine others might have the same problem.

Ref: https://github.com/rubocop-hq/ruby-style-guide#file-classes

File: app/domain/util/stubs/deep_double.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
+module Util
+  module Stubs
+    class DeepDouble
```
Comment (micahlee) [inline app/domain/util/stubs/deep_double.rb]: This could benefit from some documentation comments describing what it is, why it exists, and how to use it.

**Comment by jonahx** [comment]
> but it could be helpful if we could create a tech debt issue noting other existing parts of our code that would benefit from refactoring to use this, versus (I assume) using the Sequel models to fetch secret values directly.

This is a great idea.

**Pull request merged**